engine/src/main/java/io/graphqlcrud/SQLDataFetcher.java

```diff
@@ -35,34 +36,80 @@
     @Override
     public ResultSetList get(DataFetchingEnvironment environment) throws Exception {
         SQLContext ctx = environment.getContext();
+        ResultSet rs = null;
 
         GraphQLType type = environment.getParentType();
-        String mutation;
+
         if(type instanceof GraphQLObjectType) {
             String name = ((GraphQLObjectType) type).getName();
+            String fieldName =  environment.getField().getName();
+
             if(name.equals("MutationType")) {
-                mutation = buildMutation(environment);
-                Connection connection = ctx.getConnection();
-                Statement statement = connection.createStatement();
-                statement.execute(mutation);
-                LOGGER.info("SQL executed: " + mutation);
+                if(fieldName.contains("create") || fieldName.contains("update")) {
+                    if(executeMutation(environment, ctx) > 0) {
+                        rs = executeSQL(environment, ctx);
+                    } else {
+                        throw new RuntimeException(" Failed to execute : No rows affected ");
+                    }
+                } else if(fieldName.contains("delete")) {
+                    rs = executeSQL(environment,ctx);
+                    if(rs != null) {
+                        executeMutation(environment,ctx);
+                    } else {
+                        throw new RuntimeException((" Failed to execute : No rows affected "));
+                    }
+                }
+            } else if(name.equals("QueryType")) {
+                rs = executeSQL(environment,ctx);
             }
         }
+        return new ResultSetList(rs, true);
+    }
+
+    private int executeMutation(DataFetchingEnvironment environment, SQLContext ctx) throws Exception {
+        String mutation = buildMutation(environment);
+        ctx.setSqlMutation(mutation);
+        Connection connection = ctx.getConnection();
+        Statement statement = connection.createStatement();
+        int count = statement.executeUpdate(mutation);
+        LOGGER.info("SQL executed: " + mutation);
 
+        //TODO : Generated keys
+//        Value<?> key = null;
+//        try (ResultSet statementGeneratedKeys = statement.getGeneratedKeys()) {
+//            if(statementGeneratedKeys.next()) {
+//                key = (Value<?>) statementGeneratedKeys.getObject(1);
+//                ctx.setAutoGeneratedPrimaryKey(key);
+//            }
+//        }
+       return count;
+    }
+
+    private ResultSet executeSQL(DataFetchingEnvironment environment, SQLContext ctx) throws Exception {
         String sql = buildSQL(environment);
         ctx.setSQL(sql);
         LOGGER.info("SQL Executed:" + sql);
 
         ResultSet rs = null;
         Connection c = ctx.getConnection();
-        Statement stmt = c.createStatement();
+        Statement stmt = c.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,ResultSet.CONCUR_READ_ONLY);
```

Comment:
why do you need to set this? this is an expensive operation also not all jdbc drivers support this

Reply:
Oops realized that now :/ 
I was trying to get the no. of rows returned after query execution to validate delete operation.